### ISSUE TITLE
TA Comments

### ISSUE THREAD

**Issue by Rashedul** [STAT545-UBC-students/hw06-shreeramsenthi#4]
- You could add table of contents.

- Instead of doing one task, I might attempt two tasks to fulfill the requirements of the assignment. However, its your choice.

- In making function, you did a fantastic job with where you wrote functions and explore the functionality.

- Your discussion/interpretation was precise.

- Great coding style and strategies. 

- Overall good job!

**Comment by shreeramsenthi** [comment]
Thanks @Rashedul ! Just to clarify, was the 20% I lost entirely from choosing to do a deeper dive into a single task instead of doing two? I still think I accomplished the overall task of demonstrating the use of the `purrr::map*` family of functions, building functions, error handling with `stop`, and using `kable` and theme elements from `ggplot2` to clean up data visualizations.

And the table of contents is a good idea, unfortunately I use a different IDE and it doesn't seem to pick up on that option in the YAML header. I'll be sure to give it another shot next time!

But in all seriousness, thanks again for the feedback and sorry to be such a pain, I just don't want to be surprised about my mark again next time.

Cheers,
Shree

**Comment by Rashedul** [comment]
Hi @shreeramsenthi thanks for sharing your thoughts. Yes, you lost some of your grades because you skipped one task. If you can rationalize how you "accomplished the overall task" explaining how you covered the works of tasks, I am willing to revise your grade. 

Best
Rashedul

**Comment by Rashedul** [comment]
"explaining how you covered the works of **two** asks"

**Comment by shreeramsenthi** [comment]
Sounds good, and thanks for the consideration! I know you certainly aren't obliged to reconsider.

So, I think I covered the main goals of tasks 2 and 5. Task 2 was to build functions that the built-in functions don't accomplish easily and have practical value. It's pretty obvious that I made some functions, but I'd argue they are useful as well since all the model building (`lm`, `glm`, etc.) and summarizing (`Anova`, `confint`, `tidy`, etc.) functions don't have an obvious way of handling multiple models easily, and my set of functions facilitate exactly that. As for actual utility, I am using it in my research now for exploratory analyses and hypothesis generation.

Task 5 on the other hand was all about getting practice with lists and `purrr::map`. Although the Trump Tweets subtask aslo deals with `stringr` and the Github Users subtask requires dealing with json files, I would argue the unifying theme is learning to use `purrr` to handle lists. The core idea of all the functions I built was dealing with lists of models, formulas, and tibbles of summary stats using `purrr`. While my lists only got really messy in a handful of places (like when I was throwing around lists of tibbles of augmented data and summary stats), I think I demonstrated a pretty diverse set of the `map*` family of functions and wound up manipulating lots of different types of data stored in lists.

And that's about it. Sorry that got a little long-winded!
